src/app/ClusterStateCache.cpp

```diff
@@ -417,8 +405,9 @@ void ClusterStateCache::GetSortedFilters(std::vector<std::pair<DataVersionFilter
                 }
                 else
                 {
-                    System::PacketBufferTLVReader bufReader;
-                    bufReader.Init(attributeIter.second.Get<System::PacketBufferHandle>().Retain());
+                    TLV::TLVReader bufReader;
+                    bufReader.Init(attributeIter.second.Get<Platform::ScopedMemoryBuffer<uint8_t>>().Get(),
```

Comment:
See my comment below, this could be computed dynamically, avoiding the need to store an extra arg inside the scoped buffer.

Reply:
follow your previous suggestion and use the wrapper to have scoped buffer and mSize, thanks